manager/controllers/app/blueprint_mgr.go

```diff
@@ -72,27 +82,152 @@ func (r *PlotterReconciler) GenerateBlueprints(instances []ModuleInstanceSpec, p
 // the paths for accessing them are included in the blueprint.
 // The credentials themselves are not included in the blueprint.
 func (r *PlotterReconciler) GenerateBlueprint(instances []ModuleInstanceSpec,
-	clusterName string,
-	plotter *fapp.Plotter) fapp.BlueprintSpec {
+	clusterName string, plotter *fapp.Plotter, services Services, ingressMap map[string][]string) fapp.BlueprintSpec {
 	spec := fapp.BlueprintSpec{
 		Cluster:          clusterName,
 		ModulesNamespace: plotter.Spec.ModulesNamespace,
 		Modules:          map[string]fapp.BlueprintModule{},
 		Application: &fapp.ApplicationDetails{
 			WorkloadSelector: plotter.Spec.Selector.WorkloadSelector,
+			Namespaces:       plotter.Spec.Selector.Namespaces,
+			IPBlocks:         plotter.Spec.Selector.IPBlocks,
 			Context:          plotter.Spec.AppInfo,
 		},
 	}
 	// Create the map that contains BlueprintModules
 	for ind := range instances {
-		modulename := instances[ind].Module.Name
-		instanceName := modulename
-		if instances[ind].Scope == fapp.Asset {
-			// Need unique name for each module
-			// if the module scope is one per asset then concat the id of the asset to it
-			instanceName = utils.CreateStepName(modulename, instances[ind].Module.AssetIDs[0])
+		var assetID string
+		if len(instances[ind].Module.AssetIDs) > 0 {
+			assetID = instances[ind].Module.AssetIDs[0]
+		}
+		instanceName := mngrUtils.CreateStepName(instances[ind].Module.Name, assetID, instances[ind].Scope)
+		releaseName := mngrUtils.GetReleaseName(mngrUtils.GetApplicationNameFromLabels(plotter.Labels),
+			mngrUtils.GetFybrikApplicationUUIDfromAnnotations(plotter.Annotations), instanceName)
+		moduleKey := UniqueReleaseName(instances[ind].ClusterName, releaseName)
+		isEndpoint := services[moduleKey].IsEndpoint
+		// connections from the module to its arguments (other modules or data locations)
+		urls := []string{}
```

Comment:
ExternalServices should be appended as well to urls.

Reply:
it's your commit https://github.com/fybrik/fybrik/pull/2053/files#diff-64d2950d3190f08f3f05255b74aa9c93798c326f0601f3c2241cf1b9353654a2R106 :-)

Reply:
In my commit there was no such field, so I could not add it.